ogre_mesh_viewer.py

```diff
@@ -609,10 +646,18 @@ def shutdown(self):
     import argparse
 
     parser = argparse.ArgumentParser(description="Ogre Mesh Viewer")
-    parser.add_argument("infile", help="path to a ogre .mesh, ogre .scene or any format supported by assimp")
-    parser.add_argument("-c", "--rescfg", help="path to the resources.cfg")
+    parser.add_argument("infile", help="path to a ogre .mesh, ogre .scene or any format supported by assimp", nargs='?')
+    parser.add_argument("-c", "--rescfg", help="path to the resources.cfg", required=False)
+    parser.set_defaults(infile='')
     args = parser.parse_args()
-    app = MeshViewer(args.infile, args.rescfg)
+
+    if args.infile == "":
+        root = tk.Tk()
+        root.withdraw()
+        infile = filedialog.askopenfilename(title="Open", filetypes=[("OGRE Mesh files", "*.mesh"), ("OGRE Scene files", "*.scene")])
+        app = MeshViewer(infile, args.rescfg)
+    else:
+        app = MeshViewer(args.infile, args.rescfg)
```

Comment:
DRY: just use `infile` in both cases and assign args.value to it if it is not empty